technique/sans/instrument.py

```diff
@@ -25,62 +25,183 @@ def _get_times(times):
+                '1WB', '2WB', '3WB', '4WB', '5WB', '6WB', '7WB',
+                '8WB', '9WB', '10WB', '11WB', '12WB', '13WB', '14WB',
+                '1WT', '2WT', '3WT', '4WT', '5WT', '6WT', '7WT',
+                '8WT', '9WT', '10WT', '11WT', '12WT', '13WT', '14WT',
+                '1GT', '2GT', '3GT', '4GT', '5GT', '6GT', '7GT', '8GT', '9GT',
+                '10GT', '11GT', '12GT', '2GB', '3GB', '4GB', '5GB', '6GB', '7GB',
+                '8GB', '9GB', '10GB', '11GB']
+
+    _poslist_dls = []
 
     def __init__(self):
+        self._poslist = self.get_pv("LKUP:SAMPLE:POSITIONS").split()
         self.setup_sans = self.setup_dae_event
         self.setup_trans = self.setup_dae_transmission
 
     def method_iterator(self):
         """Iterate through the class's public functions"""
         for method in dir(self):
             if method[0] != "_" and method not in locals() and \
-               method not in self._block_accessors and \
-               callable(getattr(self, method)):
+                    method not in self._block_accessors and \
+                    callable(getattr(self, method)):
                 yield method
 
     def set_default_dae(self, mode=None, trans=False):
-        """Set the default DAE mode for SANS or TRANS measuremnts.
+        """Set the default DAE mode for SANS or TRANS measurements.
 
         Parameters
         ----------
-        mode : str or function
-          If the mode is a function, call that function to set the DAE
-          mode.  If the mode is a string, call the function whose name
-          is "setup_dae_" followed by that string.
+        mode : str
+          The mode is a string, call the function whose name
+          is "setup_dae_<mode>" followed by that string.
+          You can find a full list of setup_dae_<name> by calling
+          enumerate_dae() on the instrument.
         trans : bool
           If true, set the default transmission instead of the default
           SANS mode.
 
         """
         if mode is None:
             pass
-        elif isinstance(mode, str):
-            self.set_default_dae(
-                getattr(self, "setup_dae_" + mode),
-                trans)
         else:
-            if trans:
-                self.setup_trans = mode
-            else:
-                self.setup_sans = mode
+            try:
+                if trans:
+                    self.setup_trans = getattr(self, "setup_dae_" + mode)
+                else:
+                    self.setup_sans = getattr(self, "setup_dae_" + mode)
+            except AttributeError:
+                print(f"No dae called 'setup_dae_{mode}', "
+                      f"you can use setup_dae_custom to create custom daes")
+
+    # I don't like that users can parse any function to create DAE that's sounds bad
+    # This would give up some more control over the function that they set but would
+    # take away there freedom.
```

Comment:
I believe that I'm the only person to every to use the ability to create custom DAEs on the fly.  Since I'm no longer on a beam line, this feels like a simpler replacement that covers most of the cases.